lightning/src/ln/channelmonitor.rs

```diff
@@ -628,89 +599,31 @@ pub struct ChannelMonitor<ChanSigner: ChannelKeys> {
-	// another bumped aggregate tx to comply with RBF rules. We may have multiple claiming txn in the flight for the
-	// same set of outpoints. One of the outpoints may be spent by a transaction not issued by us. That's why at
-	// block connection we scan all inputs and if any of them is among a set of a claiming request we test for set
-	// equality between spending transaction and claim request. If true, it means transaction was one our claiming one
-	// after a security delay of 6 blocks we remove pending claim request. If false, it means transaction wasn't and
-	// we need to regenerate new claim request we reduced set of stil-claimable outpoints.
-	// Key is identifier of the pending claim request, i.e the txid of the initial claiming transaction generated by
-	// us and is immutable until all outpoint of the claimable set are post-anti-reorg-delay solved.
-	// Entry is cache of elements need to generate a bumped claiming transaction (see ClaimTxBumpMaterial)
-	#[cfg(test)] // Used in functional_test to verify sanitization
-	pub pending_claim_requests: HashMap<Sha256dHash, ClaimTxBumpMaterial>,
-	#[cfg(not(test))]
-	pending_claim_requests: HashMap<Sha256dHash, ClaimTxBumpMaterial>,
-
-	// Used to link outpoints claimed in a connected block to a pending claim request.
-	// Key is outpoint than monitor parsing has detected we have keys/scripts to claim
-	// Value is (pending claim request identifier, confirmation_block), identifier
-	// is txid of the initial claiming transaction and is immutable until outpoint is
-	// post-anti-reorg-delay solved, confirmaiton_block is used to erase entry if
-	// block with output gets disconnected.
-	#[cfg(test)] // Used in functional_test to verify sanitization
-	pub claimable_outpoints: HashMap<BitcoinOutPoint, (Sha256dHash, u32)>,
-	#[cfg(not(test))]
-	claimable_outpoints: HashMap<BitcoinOutPoint, (Sha256dHash, u32)>,
-
 	// Used to track onchain events, i.e transactions parts of channels confirmed on chain, on which
 	// we have to take actions once they reach enough confs. Key is a block height timer, i.e we enforce
 	// actions when we receive a block with given height. Actions depend on OnchainEvent type.
 	onchain_events_waiting_threshold_conf: HashMap<u32, Vec<OnchainEvent>>,
 
+	#[cfg(test)]
+	pub onchain_tx_handler: OnchainTxHandler,
+	#[cfg(not(test))]
+	onchain_tx_handler: OnchainTxHandler,
+
 	// We simply modify last_block_hash in Channel's block_connected so that serialization is
 	// consistent but hopefully the users' copy handles block_connected in a consistent way.
 	// (we do *not*, however, update them in insert_combine to ensure any local user copies keep
 	// their last_block_hash from its state and not based on updated copies that didn't run through
 	// the full block_connected).
 	pub(crate) last_block_hash: Sha256dHash,
-	secp_ctx: Secp256k1<secp256k1::All>, //TODO: dedup this a bit...
+	secp_ctx: Secp256k1<secp256k1::All>,
```

Comment:
Why? We should still think about deduping the secp contexts especially in a ManyChannelMonitor.

Reply:
Ooops, I tried at first to throw it entirely in OnchainTxHandler IIIRC. That's said, after moving local commitment txn construction there too, secp state should be the matter of the Signer